scripts/create_compdb.py

```diff
@@ -23,145 +23,16 @@
 
-# Load compiler flags. We do this first in order to fail fast if not run from
-# the workspace root.
-print("Reading the arguments to use...")
-try:
-    with open("compile_flags.txt") as flag_file:
-        arguments = [line.strip() for line in flag_file]
-except FileNotFoundError:
-    sys.exit(Path(sys.argv[0]).name + " must be run from the project root")
 
-# Prepend the `clang` executable path to the arguments that looks into our
-# downloaded Clang toolchain.
-arguments = ["clang++"] + arguments
-
-print("Building compilation database...")
-
-# Find all of the C++ source files that we expect to compile cleanly as
-# stand-alone files. This is a bit simpler than scraping the actual compile
-# actions and allows us to directly index header-only libraries easily and
-# pro-actively index the specific headers in the project.
-source_files_query = subprocess.check_output(
-    [
-        bazel,
-        "query",
-        "--keep_going",
-        "--output=location",
-        # Workaround for https://github.com/bazelbuild/bazel/issues/8900
-        "--incompatible_display_source_file_location",
-        'filter(".*\\.(h|cpp|cc|c|cxx)$", kind("source file", deps(//...)))',
-    ],
-    stderr=subprocess.DEVNULL,
-    universal_newlines=True,
-)
-source_files = [
-    Path(line.split(":")[0]) for line in source_files_query.splitlines()
-]
-
-# Filter into the Carbon source files that we'll find directly in the
-# workspace, and LLVM source files that need to be mapped through the merged
-# LLVM tree in Bazel's execution root.
-carbon_files = [
-    f.relative_to(directory)
-    for f in source_files
-    if f.parts[: len(directory.parts)] == directory.parts
-]
-llvm_files = [
-    Path("bazel-execroot/external").joinpath(
-        *f.parts[f.parts.index("llvm-project") :]
-    )
-    for f in source_files
-    if "llvm-project" in f.parts
-]
-print(
-    "Found %d Carbon source files and %d LLVM source files..."
-    % (len(carbon_files), len(llvm_files))
-)
-
-# Now collect the generated file labels.
-# cc_proto_library generates files, but they aren't seen with "generated file".
-generated_file_labels = subprocess.check_output(
-    [
-        bazel,
-        "query",
-        "--keep_going",
-        "--output=label",
-        (
-            'filter(".*\\.(h|cpp|cc|c|cxx|def|inc)$",'
-            'kind("generated file", deps(//...)))'
-            " union "
-            'kind("cc_proto_library", deps(//...))'
-        ),
-    ],
-    stderr=subprocess.DEVNULL,
-    universal_newlines=True,
-).splitlines()
-print("Found %d generated files..." % (len(generated_file_labels),))
-
-# Directly build these labels so that indexing can find them. Allow this to
-# fail in case there are build errors in the client, and just warn the user
-# that they may be missing generated files.
-print("Building the generated files so that tools can find them...")
-subprocess.check_call([bazel, "build", "--keep_going"] + generated_file_labels)
```

Comment:
I think we need to keep building the generated files before running this? At least, the docs seem to indicate you still need to do a successful build to get generated files?

Reply:
Done, I don't think I was running into issues here because the toolchain doesn't benefit from this.